src/test/java/org/springframework/retry/backoff/BackOffPolicyBuilderTests.java

```diff
@@ -109,4 +110,60 @@ public void shouldCreateExponentialRandomBackOff() {
 		assertThat(new DirectFieldAccessor(policy).getPropertyValue("sleeper")).isEqualTo(mockSleeper);
 	}
 
+	@Test
+	public void shouldCreateExponentialRandomBackOffWhenProvidedRandomSupplier() {
+		Sleeper mockSleeper = mock(Sleeper.class);
+		BackOffPolicy backOffPolicy = BackOffPolicyBuilder.newBuilder()
+			.delay(10000)
+			.maxDelay(100000)
+			.multiplier(10)
+			.randomSupplier(() -> true)
```

Comment:
Can this also be like `random(true)`?
For consistency with the rest of properties you set here.

Reply:
That test is already present. The purpose of this test is to test the scenario where random is provided via expression whereas other properties are values.